base:
  id: 1
  type: 01-BASE
  description: Base Panel
  visual_meshes:
  - url: package://aist_object_database/meshes/01-BASE.stl
    pose: [0.2, 0.0, 0.12, 0.0, 180.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: BOX
    dimensions: [0.2, 0.008, 0.12]
    pose: [0.1, 0.004, 0.06, 0.0, 0.0, 0.0]
  - type: BOX
    dimensions: [0.02, 0.02, 0.02]
    pose: [0.1, 0.012, 0.1, 0.0, 0.0, 0.0]
  subframes:
    big_hole_1: [0.013, 0.0, 0.076, 90.0, 0.0, -180.0]
    big_hole_2: [0.076, 0.0, 0.076, 90.0, 0.0, -180.0]
    big_hole_3: [0.124, 0.0, 0.076, 90.0, 0.0, -180.0]
    big_hole_4: [0.187, 0.0, 0.076, 90.0, 0.0, -180.0]
    under_outer_hole: [0.01, 0.0, 0.01, 90.0, 0.0, -90.0]
    corner_1: [0.0, 0.01, 0.0, 90.0, 0.0, -180.0]
    corner_2: [0.2, 0.008, 0.0, 90.0, 0.0, -180.0]
    corner_3: [0.2, 0.008, 0.12, 90.0, 0.0, -180.0]
    corner_4: [0.0, 0.008, 0.12, 90.0, 0.0, -180.0]
    screw_hole_panel1_1: [0.006, 0.008, 0.030, 90.0, 0.0, -180.0]
    screw_hole_panel1_2: [0.11, 0.008, 0.030, 90.0, 0.0, -180.0]
    screw_hole_panel1_3: [0.006, 0.0, 0.030, 90.0, 0.0, -180.0]
    screw_hole_panel1_4: [0.11, 0.0, 0.030, 90.0, 0.0, -180.0]
    screw_hole_panel2_1: [0.131, 0.008, 0.028, 90.0, 0.0, -180.0]
    screw_hole_panel2_2: [0.179, 0.008, 0.028, 90.0, 0.0, -180.0]
    screw_hole_panel2_3: [0.131, 0.0, 0.028, 90.0, 0.0, -180.0]
    screw_hole_panel2_4: [0.179, 0.0, 0.028, 90.0, 0.0, -180.0]
    terminal_top: [0.1, 0.028, 0.099, -90.0, -90.0, 0.0]
    cable_pin1: [0.092, 0.028, 0.108, -75.0, -90.0, 0.0]
    cable_pin2: [0.092, 0.028, 0.092, -75.0, -90.0, 0.0]
    cable_pin3: [0.108, 0.028, 0.092, -105.0, -90.0, 0.0]
    cable_pin4: [0.108, 0.028, 0.108, -105.0, -90.0, 0.0]
  grasp_points:
    default_grasp: [0.1, 0.0, 0.06, -90.0, -0.0, -0.0]
    terminal_grasp: [0.1, 0.008, 0.1, -90.0, 90.0, 0.0]
    big_holes_grasp: [0.1, 0.0, 0.076, -90.0, 90.0, 0.0]
panel_motor:
  id: 2
  type: 02-PANEL
  description: Motor fixing plate
  visual_meshes:
  - url: package://aist_object_database/meshes/02-PANEL.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: BOX
    dimensions: [0.06, 0.06, 0.005]
    pose: [0.03, 0.03, 0.0025, 0.0, 0.0, 0.0]
  - type: BOX
    dimensions: [0.06, 0.005, 0.025]
    pose: [0.03, 0.0025, 0.0125, 0.0, 0.0, 0.0]
  subframes:
    bottom_corner_1: [0.0, 0.0, 0.0, 90.0, 0.0, -180.0]
    bottom_corner_2: [0.06, 0.0, 0.0, 90.0, 0.0, -180.0]
    back_hole: [0.03, 0.046, 0.005, -0.0, 0.0, -90.0]
    back_hole_bottom: [0.03, 0.034, 0.005, -0.0, 0.0, 90.0]
    center: [0.03, 0.03, 0.0, 180.0, -0.0, -90.0]
    bottom_screw_hole_aligner_1: [0.006, 0.0, 0.015, 90.0, 0.0, -180.0]
    bottom_screw_hole_aligner_2: [0.054, 0.0, 0.015, 90.0, 0.0, -180.0]
    bottom_screw_hole_1: [0.006, 0.005, 0.015, 90.0, 0.0, -180.0]
    bottom_screw_hole_2: [0.054, 0.005, 0.015, 90.0, 0.0, -180.0]
    motor_screw_hole_1: [0.03775, 0.051423, 0.0, 180.0, -0.0, 90.0]
    motor_screw_hole_2: [0.0455, 0.038, 0.0, 180.0, -0.0, 90.0]
    motor_screw_hole_3: [0.03775, 0.024577, 0.0, 180.0, -0.0, 90.0]
    motor_screw_hole_4: [0.02225, 0.024577, 0.0, 180.0, -0.0, 90.0]
    motor_screw_hole_5: [0.01415, 0.038, 0.0, 180.0, -0.0, 90.0]
    motor_screw_hole_6: [0.02225, 0.051423, 0.0, 180.0, -0.0, 90.0]
  grasp_points:
    default_grasp: [0.032, 0.0025, 0.01, -0.0, -0.0, 0.0]
    grasp_1: [0.032, 0.0025, 0.01, -0.0, 60.0, 180.0]
    grasp_2: [0.032, 0.0025, 0.01, -0.0, 30.0, 180.0]
    grasp_3: [0.032, 0.0025, 0.01, -0.0, 0.0, 180.0]
    grasp_4: [0.032, 0.0025, 0.01, -0.0, -30.0, 180.0]
    grasp_5: [0.032, 0.0025, 0.01, -0.0, -60.0, 180.0]
    grasp_6: [0.033, 0.04, 0.0025, -90.0, 0.0, -90.0]
    grasp_7: [0.033, 0.04, 0.0025, -90.0, 0.0, -60.0]
    grasp_8: [0.033, 0.04, 0.0025, -90.0, 0.0, -30.0]
    grasp_9: [0.033, 0.04, 0.0025, 90.0, 0.0, 180.0]
    grasp_10: [0.033, 0.04, 0.0025, -90.0, 0.0, 30.0]
    grasp_11: [0.033, 0.04, 0.0025, -90.0, 0.0, 60.0]
    grasp_12: [0.033, 0.04, 0.0025, -90.0, 0.0, 90.0]
panel_bearing:
  id: 3
  type: 03-PANEL2
  description: Output shaft fixing plate
  visual_meshes:
  - url: package://aist_object_database/meshes/03-PANEL2.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: BOX
    dimensions: [0.116, 0.09, 0.005]
    pose: [0.058, 0.045, 0.0025, 0.0, 0.0, 0.0]
  - type: BOX
    dimensions: [0.116, 0.005, 0.025]
    pose: [0.058, 0.0025, 0.0125, 0.0, 0.0, 0.0]
  subframes:
    bottom_corner_1: [0.0, 0.0, 0.0, 90.0, 0.0, -180.0]
    bottom_corner_2: [0.116, 0.0, 0.0, 90.0, 0.0, -180.0]
    bottom_screw_hole_aligner_1: [0.006, 0.0, 0.015, 90.0, 0.0, -180.0]
    bottom_screw_hole_aligner_2: [0.11, 0.0, 0.015, 90.0, 0.0, -180.0]
    bottom_screw_hole_1: [0.006, 0.005, 0.015, 90.0, 0.0, -180.0]
    bottom_screw_hole_2: [0.11, 0.005, 0.015, 90.0, 0.0, -180.0]
    front_hole: [0.035, 0.045, 0.0, 180.0, -0.0, -90.0]
    center: [0.058, 0.045, 0.0, 180.0, -0.0, -90.0]
    pulley_ridge_bottom: [0.095, 0.045, 0.0, 180.0, -0.0, 90.0]
    pulley_ridge_middle: [0.095, 0.0625, 0.0, 180.0, -0.0, 90.0]
    pulley_ridge_top: [0.095, 0.08, 0.0, 180.0, -0.0, 90.0]
  grasp_points:
    default_grasp: [0.058, 0.0025, 0.01, -0.0, -0.0, 0.0]
    grasp_1: [0.035, 0.06, 0.0025, -90.0, 0.0, -30.0]
    grasp_2: [0.035, 0.06, 0.0025, -90.0, 0.0, -0.0]
    grasp_3: [0.035, 0.06, 0.0025, -90.0, 0.0, 30.0]
    grasp_4: [0.035, 0.06, 0.0025, -90.0, 0.0, 60.0]
    grasp_5: [0.035, 0.06, 0.0025, -90.0, 0.0, 90.0]
    grasp_6: [0.058, 0.06, 0.0025, -90.0, 0.0, -30.0]
    grasp_7: [0.058, 0.06, 0.0025, 90.0, 0.0, 180.0]
    grasp_8: [0.058, 0.06, 0.0025, -90.0, 0.0, 30.0]
    grasp_9: [0.081, 0.06, 0.0025, -90.0, 0.0, -90.0]
    grasp_10: [0.081, 0.06, 0.0025, -90.0, 0.0, -60.0]
    grasp_11: [0.081, 0.06, 0.0025, -90.0, 0.0, -30.0]
    grasp_12: [0.081, 0.06, 0.0025, -90.0, 0.0, -0.0]
    grasp_13: [0.081, 0.06, 0.0025, -90.0, 0.0, 30.0]
    grasp_14: [0.081, 0.0025, 0.01, -0.0, 60.0, 180.0]
    grasp_15: [0.081, 0.0025, 0.01, -0.0, 30.0, 180.0]
    grasp_16: [0.081, 0.0025, 0.01, -0.0, 0.0, 180.0]
    grasp_17: [0.081, 0.0025, 0.01, -0.0, -30.0, 180.0]
    grasp_18: [0.081, 0.0025, 0.01, -0.0, -60.0, 180.0]
    grasp_19: [0.058, 0.0025, 0.01, -0.0, 60.0, 180.0]
    grasp_20: [0.058, 0.0025, 0.01, -0.0, 30.0, 180.0]
    grasp_21: [0.058, 0.0025, 0.01, -0.0, 0.0, 180.0]
    grasp_22: [0.058, 0.0025, 0.01, -0.0, -30.0, 180.0]
    grasp_23: [0.058, 0.0025, 0.01, -0.0, -60.0, 180.0]
    grasp_24: [0.035, 0.0025, 0.01, -0.0, 60.0, 180.0]
    grasp_25: [0.035, 0.0025, 0.01, -0.0, 30.0, 180.0]
    grasp_26: [0.035, 0.0025, 0.01, -0.0, 0.0, 180.0]
    grasp_27: [0.035, 0.0025, 0.01, -0.0, -30.0, 180.0]
    grasp_28: [0.035, 0.0025, 0.01, -0.0, -60.0, 180.0]
motor:
  id: 4
  type: 04_37D-GEARMOTOR-50-70
  description: Geared motor (Gear ratio 1:70)
  visual_meshes:
  - url: package://aist_object_database/meshes/04_37D-GEARMOTOR-50-70.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.055, 0.0190]
    pose: [0.020, 0.0, -0.0075, 0.0, 0.0, 0.0]
  - type: CYLINDER
    dimensions: [0.022, 0.003]
    pose: [0.020, -0.007, 0.031, 0.0, 0.0, 0.0]
  subframes:
    center: [0.016, 0.0, 0.0, 180.0, -0.0, -90.0]
    tip: [0.02, -0.007, 0.042, 180.0, -0.0, -90.0]
    inserted_02: [0.02, -0.007, 0.021, 180.0, -0.0, -90.0]
    black_cable_connector: [0.011, -0.006, -0.04, 180.0, -0.0, -90.0]
    red_cable_connector: [0.029, 0.006, -0.04, 180.0, -0.0, -90.0]
  grasp_points:
    grasp_1: [0.019, 0.0, 0.0, 180.0, 90.0, 0.0]
    grasp_2: [0.019, 0.0, 0.0, 150.0, 90.0, 0.0]
    grasp_3: [0.019, 0.0, 0.0, 120.0, 90.0, 0.0]
    grasp_4: [0.019, 0.0, 0.0, 90.0, 90.0, 0.0]
    grasp_5: [0.019, 0.0, 0.0, 60.0, 90.0, 0.0]
    grasp_6: [0.019, 0.0, 0.0, 30.0, 90.0, 0.0]
    grasp_7: [0.019, 0.0, 0.0, -0.0, 90.0, 0.0]
    grasp_8: [0.019, 0.0, 0.0, -30.0, 90.0, 0.0]
    grasp_9: [0.019, 0.0, 0.0, -60.0, 90.0, 0.0]
    grasp_10: [0.019, 0.0, 0.0, -90.0, 90.0, 0.0]
    grasp_11: [0.019, 0.0, 0.0, -120.0, 90.0, 0.0]
    grasp_12: [0.019, 0.0, 0.0, -150.0, 90.0, 0.0]
motor_pulley:
  id: 5
  type: 05_MBRFA30-2-P6
  description: Pulley for Round Belt (4mm) - Setscrew P.D. 30mm
  visual_meshes:
  - url: package://aist_object_database/meshes/05_MBRFA30-2-P6.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.010, 0.016]
    pose: [0.0, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    center: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [0.0, 0.0, 0.0, 180.0, 90.0, 0.0]
    grasp_2: [0.0, 0.0, 0.0, -90.0, 60.0, 90.0]
    grasp_3: [0.0, 0.0, 0.0, -90.0, 30.0, 90.0]
    grasp_4: [0.0, 0.0, 0.0, -90.0, 0.0, 90.0]
    grasp_5: [0.0, 0.0, 0.0, -90.0, -30.0, 90.0]
    grasp_6: [0.0, 0.0, 0.0, -90.0, -60.0, 90.0]
    grasp_7: [0.0, 0.0, 0.0, -0.0, -90.0, 0.0]
    grasp_8: [0.0, 0.0, 0.0, 90.0, -60.0, -90.0]
    grasp_9: [0.0, 0.0, 0.0, 90.0, -30.0, -90.0]
    grasp_10: [0.0, 0.0, 0.0, 90.0, -0.0, -90.0]
    grasp_11: [0.0, 0.0, 0.0, 90.0, 30.0, -90.0]
    grasp_12: [0.0, 0.0, 0.0, 90.0, 60.0, -90.0]
    grasp_13: [0.0, 0.0, 0.0, -0.0, 90.0, 0.0]
    grasp_14: [0.0, 0.0, 0.0, -90.0, 60.0, -90.0]
    grasp_15: [0.0, 0.0, 0.0, -90.0, 30.0, -90.0]
    grasp_16: [0.0, 0.0, 0.0, -90.0, 0.0, -90.0]
    grasp_17: [0.0, 0.0, 0.0, -90.0, -30.0, -90.0]
    grasp_18: [0.0, 0.0, 0.0, -90.0, -60.0, -90.0]
    grasp_19: [0.0, 0.0, 0.0, 180.0, -90.0, 0.0]
    grasp_20: [0.0, 0.0, 0.0, 90.0, -60.0, 90.0]
    grasp_21: [0.0, 0.0, 0.0, 90.0, -30.0, 90.0]
    grasp_22: [0.0, 0.0, 0.0, 90.0, -0.0, 90.0]
    grasp_23: [0.0, 0.0, 0.0, 90.0, 30.0, 90.0]
    grasp_24: [0.0, 0.0, 0.0, 90.0, 60.0, 90.0]
bearing:
  id: 7
  type: 07_SBARB6200ZZ-30
  description: Bearings with Housings (Double Bearings)
  visual_meshes:
  - url: package://aist_object_database/meshes/07_SBARB6200ZZ_30.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.008, 0.027]
    pose: [-0.004, 0.0, 0.0, 0.0, 90.0, 0.0]
  - type: CYLINDER
    dimensions: [0.022, 0.018]
    pose: [-0.019, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    front_hole: [0.0, 0.0, 0.0, -90.0, 0.0, -90.0]
    back_hole: [-0.03, 0.0, 0.0, 90.0, 0.0, -90.0]
    back_tip: [-0.03, 0.0, 0.0, -90.0, 0.0, -90.0]
    inserted: [-0.008, 0.0, 0.0, -90.0, 0.0, -90.0]
    screw_hole_1: [0.0, 0.0, -0.022, -90.0, 0.0, -90.0]
    screw_hole_2: [0.0, 0.022, 0.0, -90.0, 0.0, -90.0]
    screw_hole_3: [0.0, 0.0, 0.022, -90.0, 0.0, -90.0]
    screw_hole_4: [0.0, -0.022, 0.0, -90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [-0.01, 0.0, 0.0, 180.0, 90.0, 0.0]
    grasp_2: [-0.01, 0.0, 0.0, -90.0, 60.0, 90.0]
    grasp_3: [-0.01, 0.0, 0.0, -90.0, 30.0, 90.0]
    grasp_4: [-0.01, 0.0, 0.0, -90.0, 0.0, 90.0]
    grasp_5: [-0.01, 0.0, 0.0, -90.0, -30.0, 90.0]
    grasp_6: [-0.01, 0.0, 0.0, -90.0, -60.0, 90.0]
    grasp_7: [-0.01, 0.0, 0.0, -0.0, -90.0, 0.0]
    grasp_8: [-0.01, 0.0, 0.0, 90.0, -60.0, -90.0]
    grasp_9: [-0.01, 0.0, 0.0, 90.0, -30.0, -90.0]
    grasp_10: [-0.01, 0.0, 0.0, 90.0, -0.0, -90.0]
    grasp_11: [-0.01, 0.0, 0.0, 90.0, 30.0, -90.0]
    grasp_12: [-0.01, 0.0, 0.0, 90.0, 60.0, -90.0]
shaft:
  id: 8
  type: 08_KZAF1075NA4WA55GA20AA0
  description: Drive shaft (Straight) D10h6
  visual_meshes:
  - url: package://aist_object_database/meshes/08_KZAF1075NA4WA55GA20AA0.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.075, 0.005]
    pose: [0.0375, 0.0, 0.0, 0.0, 89.99, 0.0]
  subframes:
    screw_hole: [0.0, 0.0, 0.0, 180.0, 90.0, 0.0]
    front_tip: [0.075, 0.0, 0.0, 90.0, 0.0, -90.0]
    inserted: [0.026, 0.0, 0.0, -90.0, 0.0, -90.0]
    back_tip: [0.0, 0.0, 0.0, -90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [0.06, 0.0, 0.0, -90.0, 0.0, 180.0]
    grasp_2: [0.06, 0.0, 0.0, -60.0, 0.0, 180.0]
    grasp_3: [0.06, 0.0, 0.0, -30.0, 0.0, 180.0]
    grasp_4: [0.06, 0.0, 0.0, -0.0, 0.0, 180.0]
    grasp_5: [0.06, 0.0, 0.0, 30.0, 0.0, -180.0]
    grasp_6: [0.06, 0.0, 0.0, 60.0, -0.0, -180.0]
    grasp_7: [0.06, 0.0, 0.0, 90.0, -0.0, -180.0]
    grasp_8: [0.06, 0.0, 0.0, 120.0, -0.0, 180.0]
    grasp_9: [0.06, 0.0, 0.0, 150.0, -0.0, 180.0]
    grasp_10: [0.06, 0.0, 0.0, 180.0, -0.0, 180.0]
    grasp_11: [0.06, 0.0, 0.0, -150.0, -0.0, 180.0]
    grasp_12: [0.06, 0.0, 0.0, -120.0, 0.0, 180.0]
end_cap:
  id: 9
  type: 09_EDCS10
  description: End Cap for Shaft
  visual_meshes:
  - url: package://aist_object_database/meshes/09_EDCS10.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.005, 0.007]
    pose: [0.0025, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    inner_surface: [0.002, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_hole: [0.0, 0.0, 0.0, -90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [0.004, 0.0, 0.0, 180.0, 90.0, 0.0]
    grasp_2: [0.004, 0.0, 0.0, -90.0, 60.0, 90.0]
    grasp_3: [0.004, 0.0, 0.0, -90.0, 30.0, 90.0]
    grasp_4: [0.004, 0.0, 0.0, -90.0, 0.0, 90.0]
    grasp_5: [0.004, 0.0, 0.0, -90.0, -30.0, 90.0]
    grasp_6: [0.004, 0.0, 0.0, -90.0, -60.0, 90.0]
    grasp_7: [0.004, 0.0, 0.0, -0.0, -90.0, 0.0]
    grasp_8: [0.004, 0.0, 0.0, 90.0, -60.0, -90.0]
    grasp_9: [0.004, 0.0, 0.0, 90.0, -30.0, -90.0]
    grasp_10: [0.004, 0.0, 0.0, 90.0, -0.0, -90.0]
    grasp_11: [0.004, 0.0, 0.0, 90.0, 30.0, -90.0]
    grasp_12: [0.004, 0.0, 0.0, 90.0, 60.0, -90.0]
    grasp_13: [0.0, 0.0, 0.0, -0.0, 90.0, 0.0]
    grasp_14: [0.0, 0.0, 0.0, -90.0, 60.0, -90.0]
    grasp_15: [0.0, 0.0, 0.0, -90.0, 30.0, -90.0]
    grasp_16: [0.0, 0.0, 0.0, -90.0, 0.0, -90.0]
    grasp_17: [0.0, 0.0, 0.0, -90.0, -30.0, -90.0]
    grasp_18: [0.0, 0.0, 0.0, -90.0, -60.0, -90.0]
    grasp_19: [0.0, 0.0, 0.0, 180.0, -90.0, 0.0]
    grasp_20: [0.0, 0.0, 0.0, 90.0, -60.0, 90.0]
    grasp_21: [0.0, 0.0, 0.0, 90.0, -30.0, 90.0]
    grasp_22: [0.0, 0.0, 0.0, 90.0, -0.0, 90.0]
    grasp_23: [0.0, 0.0, 0.0, 90.0, 30.0, 90.0]
    grasp_24: [0.0, 0.0, 0.0, 90.0, 60.0, 90.0]
bearing_spacer:
  id: 10
  type: 10_CLBPS10-17-4
  description: Bearing Spacers For Inner Ring (output pulley)
  visual_meshes:
  - url: package://aist_object_database/meshes/10_CLBPS10_17_4.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.004, 0.0085]
    pose: [0.002, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    center: [0.002, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [0.004, 0.0, 0.0, 180.0, 90.0, 0.0]
    grasp_2: [0.004, 0.0, 0.0, -90.0, 60.0, 90.0]
    grasp_3: [0.004, 0.0, 0.0, -90.0, 30.0, 90.0]
    grasp_4: [0.004, 0.0, 0.0, -90.0, 0.0, 90.0]
    grasp_5: [0.004, 0.0, 0.0, -90.0, -30.0, 90.0]
    grasp_6: [0.004, 0.0, 0.0, -90.0, -60.0, 90.0]
    grasp_7: [0.004, 0.0, 0.0, -0.0, -90.0, 0.0]
    grasp_8: [0.004, 0.0, 0.0, 90.0, -60.0, -90.0]
    grasp_9: [0.004, 0.0, 0.0, 90.0, -30.0, -90.0]
    grasp_10: [0.004, 0.0, 0.0, 90.0, -0.0, -90.0]
    grasp_11: [0.004, 0.0, 0.0, 90.0, 30.0, -90.0]
    grasp_12: [0.004, 0.0, 0.0, 90.0, 60.0, -90.0]
output_pulley:
  id: 11
  type: 11_MBRAC60-2-10
  description: Pulley for Round Belts Clamping Type P.D. 60mm
  visual_meshes:
  - url: package://aist_object_database/meshes/11_MBRAC60-2-10.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.010, 0.031]
    pose: [0.0, 0.0, 0.0, 0.0, 90.0, 0.0]
  - type: CYLINDER
    dimensions: [0.010, 0.0125]
    pose: [0.010, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    front_hole: [0.015, 0.0, 0.0, -90.0, 0.0, -90.0]
    back_hole: [-0.005, 0.0, 0.0, 90.0, 0.0, -90.0]
    back_tip: [-0.005, 0.0, 0.0, -90.0, 0.0, -90.0]
    screw_head_1: [0.01, 0.01, -0.00875, -90.0, 90.0, 0.0]
    screw_head_2: [0.01, 0.01, 0.00875, -90.0, 90.0, 0.0]
  grasp_points:
    grasp_1: [-0.005, 0.0, 0.0, -0.0, 90.0, 0.0]
    grasp_2: [-0.005, 0.0, 0.0, -90.0, 60.0, -90.0]
    grasp_3: [-0.005, 0.0, 0.0, -90.0, 30.0, -90.0]
    grasp_4: [-0.005, 0.0, 0.0, -90.0, 0.0, -90.0]
    grasp_5: [-0.005, 0.0, 0.0, -90.0, -30.0, -90.0]
    grasp_6: [-0.005, 0.0, 0.0, -90.0, -60.0, -90.0]
    grasp_7: [-0.005, 0.0, 0.0, 180.0, -90.0, 0.0]
    grasp_8: [-0.005, 0.0, 0.0, 90.0, -60.0, 90.0]
    grasp_9: [-0.005, 0.0, 0.0, 90.0, -30.0, 90.0]
    grasp_10: [-0.005, 0.0, 0.0, 90.0, -0.0, 90.0]
    grasp_11: [-0.005, 0.0, 0.0, 90.0, 30.0, 90.0]
    grasp_12: [-0.005, 0.0, 0.0, 90.0, 60.0, 90.0]
idler_spacer:
  id: 12
  type: 12_CLBUS6-9-9.5
  description: Bearing Spacers For Inner Ring (tension pulley)
  visual_meshes:
  - url: package://aist_object_database/meshes/12_CLBUS6-9-9.5.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.0095, 0.0045]
    pose: [0.00475, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    center: [0.00475, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [0.0095, 0.0, 0.0, 180.0, 90.0, 0.0]
    grasp_2: [0.0095, 0.0, 0.0, -90.0, 60.0, 90.0]
    grasp_3: [0.0095, 0.0, 0.0, -90.0, 30.0, 90.0]
    grasp_4: [0.0095, 0.0, 0.0, -90.0, 0.0, 90.0]
    grasp_5: [0.0095, 0.0, 0.0, -90.0, -30.0, 90.0]
    grasp_6: [0.0095, 0.0, 0.0, -90.0, -60.0, 90.0]
    grasp_7: [0.0095, 0.0, 0.0, -0.0, -90.0, 0.0]
    grasp_8: [0.0095, 0.0, 0.0, 90.0, -60.0, -90.0]
    grasp_9: [0.0095, 0.0, 0.0, 90.0, -30.0, -90.0]
    grasp_10: [0.0095, 0.0, 0.0, 90.0, -0.0, -90.0]
    grasp_11: [0.0095, 0.0, 0.0, 90.0, 30.0, -90.0]
    grasp_12: [0.0095, 0.0, 0.0, 90.0, 60.0, -90.0]
    grasp_13: [0.00475, 0.0, 0.0, -90.0, 0.0, 180.0]
    grasp_14: [0.00475, 0.0, 0.0, -60.0, 0.0, 180.0]
    grasp_15: [0.00475, 0.0, 0.0, -30.0, 0.0, 180.0]
    grasp_16: [0.00475, 0.0, 0.0, -0.0, 0.0, 180.0]
    grasp_17: [0.00475, 0.0, 0.0, 30.0, 0.0, -180.0]
    grasp_18: [0.00475, 0.0, 0.0, 60.0, -0.0, -180.0]
    grasp_19: [0.00475, 0.0, 0.0, 90.0, -0.0, -180.0]
    grasp_20: [0.00475, 0.0, 0.0, 120.0, -0.0, 180.0]
    grasp_21: [0.00475, 0.0, 0.0, 150.0, -0.0, 180.0]
    grasp_22: [0.00475, 0.0, 0.0, 180.0, -0.0, 180.0]
    grasp_23: [0.00475, 0.0, 0.0, -150.0, -0.0, 180.0]
    grasp_24: [0.00475, 0.0, 0.0, -120.0, 0.0, 180.0]
idler_pulley:
  id: 13
  type: 13_MBGA30-2
  description: Idler for Round Belt - Wide
  visual_meshes:
  - url: package://aist_object_database/meshes/13_MBGA30-2.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.012, 0.016]
    pose: [0.006, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    center: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_head: [0.0, 0.016, 0.0, 90.0, 0.0, -180.0]
  grasp_points:
    grasp_1: [0.0, 0.0, 0.0, -0.0, 90.0, 0.0]
    grasp_2: [0.0, 0.0, 0.0, -90.0, 60.0, -90.0]
    grasp_3: [0.0, 0.0, 0.0, -90.0, 30.0, -90.0]
    grasp_4: [0.0, 0.0, 0.0, -90.0, 0.0, -90.0]
    grasp_5: [0.0, 0.0, 0.0, -90.0, -30.0, -90.0]
    grasp_6: [0.0, 0.0, 0.0, -90.0, -60.0, -90.0]
    grasp_7: [0.0, 0.0, 0.0, 180.0, -90.0, 0.0]
    grasp_8: [0.0, 0.0, 0.0, 90.0, -60.0, 90.0]
    grasp_9: [0.0, 0.0, 0.0, 90.0, -30.0, 90.0]
    grasp_10: [0.0, 0.0, 0.0, 90.0, -0.0, 90.0]
    grasp_11: [0.0, 0.0, 0.0, 90.0, 30.0, 90.0]
    grasp_12: [0.0, 0.0, 0.0, 90.0, 60.0, 90.0]
idler_pin:
  id: 14
  type: 14_BGPSL6-9-L30-F8
  description: Bearing Shaft Screw
  visual_meshes:
  - url: package://aist_object_database/meshes/14_BGPSL6-9-L30-F8.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.030, 0.003]
    pose: [0.015, 0.0, 0.0, 0.0, 90.0, 0.0]
  - type: CYLINDER
    dimensions: [0.002, 0.0045]
    pose: [-0.001, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    screw_tip: [0.03, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_inserted: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_head: [-0.002, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points:
    grasp_1: [0.015, 0.0, 0.0, -90.0, 0.0, 180.0]
    grasp_2: [0.015, 0.0, 0.0, -60.0, 0.0, 180.0]
    grasp_3: [0.015, 0.0, 0.0, -30.0, 0.0, 180.0]
    grasp_4: [0.015, 0.0, 0.0, -0.0, 0.0, 180.0]
    grasp_5: [0.015, 0.0, 0.0, 30.0, 0.0, -180.0]
    grasp_6: [0.015, 0.0, 0.0, 60.0, -0.0, -180.0]
    grasp_7: [0.015, 0.0, 0.0, 90.0, -0.0, -180.0]
    grasp_8: [0.015, 0.0, 0.0, 120.0, -0.0, 180.0]
    grasp_9: [0.015, 0.0, 0.0, 150.0, -0.0, 180.0]
    grasp_10: [0.015, 0.0, 0.0, 180.0, -0.0, 180.0]
    grasp_11: [0.015, 0.0, 0.0, -150.0, -0.0, 180.0]
    grasp_12: [0.015, 0.0, 0.0, -120.0, 0.0, 180.0]
m6_nut:
  id: 15
  type: 15_SLBNR6
  description: M6 Hex Nut  (Fixing for idler shaft)
  visual_meshes:
  - url: package://aist_object_database/meshes/15_SLBNR6.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.005, 0.00575]
    pose: [0.0, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    center: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points: {}
m6_washer:
  id: 16
  type: 16_SPWF6
  description: M6 Flat Washer  (Fixing for idler shaft)
  visual_meshes:
  - url: package://aist_object_database/meshes/16_SPWF6.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.0016, 0.00625]
    pose: [0.0, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    center: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points: {}
m4_screw:
  id: 17
  type: 17_SCB4-10
  description: 10mm M4 Socket Head Cap Screw  (metric coarse thread)
  visual_meshes:
  - url: package://aist_object_database/meshes/17_SCB4-10.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.010, 0.002]
    pose: [0.005, 0.0, 0.0, 0.0, 90.0, 0.0]
  - type: CYLINDER
    dimensions: [0.004, 0.0035]
    pose: [-0.002, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    screw_tip: [0.01, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_inserted: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_head: [-0.004, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points: {}
m3_screw_long:
  id: 18
  type: 18_SCB3-10
  description: 10mm M3 Socket Head Cap Screw  (metric coarse thread)
  visual_meshes:
  - url: package://aist_object_database/meshes/18_SCB3-10.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.010, 0.0015]
    pose: [0.005, 0.0, 0.0, 0.0, 90.0, 0.0]
  - type: CYLINDER
    dimensions: [0.003, 0.00275]
    pose: [-0.0015, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    screw_tip: [0.01, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_inserted: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_head: [-0.003, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points: {}
m3_screw_short:
  id: 19
  type: 19_MSSFS3-6
  description: 6mm M3 Hex Socket Set Screw  (metric coarse thread)
  visual_meshes:
  - url: package://aist_object_database/meshes/19_MSSFS3-6.stl
    pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    scale: [0.001, 0.001, 0.001]
    color: [0.2, 0.5, 0.8, 1.0]
  primitives:
  - type: CYLINDER
    dimensions: [0.006, 0.0015]
    pose: [0.0, 0.0, 0.0, 0.0, 90.0, 0.0]
  subframes:
    screw_tip: [0.01, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_inserted: [0.0, 0.0, 0.0, 90.0, 0.0, -90.0]
    screw_head: [-0.003, 0.0, 0.0, 90.0, 0.0, -90.0]
  grasp_points: {}
